src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -107,6 +108,27 @@ public async Task SendRequestBodyAsync(CancellationToken cancellationToken)
                 }
             }
 
+            public async Task SendRequestContentWithExpect100ContinueAsync(CancellationToken cancellationToken)
```

Comment:
Does this method have to live on `Http2Stream`, or can it live on `Http2Connection`? Most of the actual send/response logic is currently on `Http2Connection`, and it might be better to have this method there too. The `Http2Stream` logic is intentionally more low-level.

Similarly, below, we are looping in `ReadResponseHeadersAsync` when we get a 1xx response. I wonder if it would be better to leave `ReadResponseHeadersAsync` as-is and add logic in the caller to deal with 1xx responses, including 100 Continue.

Reply:
Related to this, it seems like we don't need a separate flag for `_abortRequestBody` if the caller handles response codes itself. Just check for a >= 300 response code and don't send the body in that case.

Reply:
I put it there because that where ReadResponseHeadersAsync() and SendRequestBodyAsync(). The 100 is specific to each request and mapping to stream seems more appropriate.